DisplayInfo/Linux/PlatformImplementation.cpp

```diff
@@ -0,0 +1,745 @@
+            : Core::SocketDatagram(false, Core::NodeId(NETLINK_KOBJECT_UEVENT, 0, 2), Core::NodeId(), 512, 1024)
+        {
+            Core::SocketDatagram::Open(Core::infinite);
+        }
+
+        /**
+         * @brief Registers a callback per unique devtype value. The specified callback will
+         * be invoked if a UDEV device with an attribute DEVTYPE=<devtype> has changed.
+         * DEVTYPE attribute can be retrieved via CLI if the user knows the path
+         * to the device:
+         *      udevadm info --path=<path_to_device>.
+         * 
+         * @param callback - Callback to be invoked on status change of a device.
+         * @param devtype - DEVTYPE attribute of a udev device.
+         * @return bool - False if a callback for a given devtype already exists.  
+         */
+        bool Register(std::function<void(const std::string&)>& callback, const std::string& devtype)
+        {
+            auto found = std::find_if(_callbacks.begin(), _callbacks.end(), [&](CallbackDescriptor& pair){
+                return pair.second == devtype;
+            });
+
+            if (found == _callbacks.end()) {
+                _callbacks.push_back({callback, devtype});
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        void Unregister(const std::string& devtype) 
+        {
+            _callbacks.erase(std::remove_if(_callbacks.begin(), _callbacks.end(), [&](CallbackDescriptor& pair) {
+                return pair.second == devtype;
+            }));
+        }
+
+        ~UdevObserverType() override
+        {
+            if(Core::SocketDatagram::IsOpen()) {
+                Core::SocketDatagram::Close(Core::infinite);
+            }
+        }
+
+        void StateChange() override {}
+
+        uint16_t SendData(VARIABLE_IS_NOT_USED uint8_t* dataFrame, VARIABLE_IS_NOT_USED const uint16_t maxSendSize) override
+        {
+            return 0;
+        }
+
+        uint16_t ReceiveData(uint8_t* dataFrame, const uint16_t receivedSize) override
+        {
+            udev_monitor_netlink_header* header = reinterpret_cast<udev_monitor_netlink_header*>(dataFrame);
+
+            if (header->filter_tag_bloom_hi == 0 && header->filter_tag_bloom_lo == 0) {
+
+                int data_index = header->properties_off / sizeof(uint8_t);
+                auto data_ptr = reinterpret_cast<char*>(&(dataFrame[data_index]));
+                auto values = GetMessageValues(data_ptr, header->properties_len);
+
+                for (auto& callback : _callbacks) {
+                    if (std::find(values.begin(), values.end(), "DEVTYPE=" + callback.second) != values.end()) {
+                        callback.first(callback.second);
+                    }
+                }
+            }
+
+            return receivedSize;
+        }
+
+    private:
+        /** 
+         * The following are structures copied from udev in order to properly parse messages coming in through
```

Comment:
Hi Haseena,
Please use the comment that I provided as well.

Reply:
done :)

Reply:
Thank you!